en/micro-integrator/docs/use-cases/examples/routing_examples/routing_based_on_headers.md

```diff
@@ -0,0 +1,141 @@
+# Routing Based on Message Headers
+    
+The example scenario uses an inventory for stocks as the back-end service and illustrates how a request message is routed to the backend based on the message header. When the router receives the stock request, it reads the request header and routes the message to the relevant mediation sequence for processing. The sequence will then forward the message to the endpoint.
+    
+## Synapse configuration
+    
+Listed below are the synapse configurations that are necessary for implementing this scenario. See the instructions on how to [build and run](#build-and-run) this example.
+
+```xml tab="Proxy Service"
+<?xml version="1.0" encoding="UTF-8"?>
+<proxy name="HeaderBasedRoutingProxy" xmlns="http://ws.apache.org/ns/synapse" transports="https http" startOnLoad="true" trace="disable">
+  <target>
+     <!-- When a request arrives the following sequence will be followed -->   
+     <inSequence>
+       <switch xmlns:ns="http://org.apache.synapse/xsd" source="get-property('transport','CustomHeader')">
+              <case regex="application/json">
+                  <log level="custom"> 
+                   <property name="'CustomHeader'" value="application/json" /> 
+                  </log> 
+                  <sequence key="sequence1" />
+              </case>
+              <case regex="text/xml">
+                  <log level="custom"> 
+                    <property name="'CustomHeader'" value="text/xml" /> 
+                  </log> 
+                  <sequence key="sequence2" />
+              </case>
+              <default>
+              <log level="custom"> 
+                  <property name="AcceptHeader" value="other" /> 
+              </log> 
+              <sequence key="sequence3" />
+              </default>
+      </switch>      
+      </inSequence>
+      <outSequence>
+          <send/>
```

Comment:
We do not need this <send/> block since all the above 3 sequences respond to the request.